api/transactions/transactions_test.go

```diff
@@ -34,24 +34,38 @@ import (
 var repo *chain.Repository
 var ts *httptest.Server
 var transaction *tx.Transaction
+var mempoolTx *tx.Transaction
 
 func TestTransaction(t *testing.T) {
 	initTransactionServer(t)
 	defer ts.Close()
+
 	getTx(t)
 	getTxReceipt(t)
-	senTx(t)
+	sendTx(t)
+	getTxWithBadId(t)
+	getReceiptWithBadId(t)
+	txWithBadHeader(t)
+	checkBlockSummaryExistsInRepoForNonExistingBlock(t)
+	getNonExistingRawTransactionWhenTxStillInMempool(t)
+	getNonPendingRawTransactionWhenTxStillInMempool(t)
+	getRawTransactionWhenTxStillInMempool(t)
+	getTransactionByIDTxNotFound(t)
+	getTransactionByIDPendingTxNotFound(t)
+	sendTxWithBadFormat(t)
+	sendTxThatCannotBeAcceptedInLocalMempool(t)
+	handleGetTransactionByIDWithBadQueryParams(t)
```

Comment:
Looks like these tests can be separated, it's a minor improvement for debugging test cases. You can always group a set of tests into one function if it's the same category.

Reply:
You mean having one Test... function for each endpoint? e.g.
```
func TestHandleSendTransaction(t *testing.T) {
  init()
  testCase1()
  testCase2()
  ...
}

func TestHandleGetTransactionByID(t *testing.T) {
  init()
  testCase1()
  testCase2()
  ...
}

func TestHandleGetTransactionReceiptByID(t *testing.T) {
  init()
  testCase1()
  testCase2()
  ...
}
```

Reply:
Yes, if it's the same category this one is a good approach https://github.com/vechain/thor/blob/master/consensus/consensus_test.go#L260

Reply:
I've added comments to distinguish among the different endpoints tests, but left the structure as is, which is the same for all the api package tests